packages/@uppy/xhr-upload/src/index.js

```diff
@@ -171,7 +171,13 @@ export default class XHRUpload extends BasePlugin {
       : Object.keys(meta) // Send along all fields by default.
 
     allowedMetaFields.forEach((item) => {
-      formData.append(item, meta[item])
+      if (Array.isArray(meta[item])) {
+        // In this case we don't need to transform `item` as a form value name,
+        // because it is already an array-like variable, for example `item[]` and it won't be overrided
```

Comment:
Would this be more correct? IIUC there's nothing that adds the brackets anywhere in our code, so it's up to the "user" (not the end-user, the user that integrates Uppy in their codebase) to add the brackets, right?
```suggestion
        // In this case we don't transform `item` to add brackets, it's up to
        // the user to add the brackets so it won't be overridden.
```

Reply:
yes, nice catch, thanks!